src/Symfony/Component/Security/Http/RateLimiter/DefaultLoginRateLimiter.php

```diff
@@ -28,11 +28,20 @@ final class DefaultLoginRateLimiter extends AbstractRequestRateLimiter
 {
     private RateLimiterFactory $globalFactory;
     private RateLimiterFactory $localFactory;
+    private string $secret;
 
-    public function __construct(RateLimiterFactory $globalFactory, RateLimiterFactory $localFactory)
+    /**
+     * @param non-empty-string $secret A secret to use for hashing the IP address and username
+     */
+    public function __construct(RateLimiterFactory $globalFactory, RateLimiterFactory $localFactory, #[\SensitiveParameter] string $secret = '')
     {
+        if (!$secret) {
```

Comment:
```suggestion
        if ('' === $secret) {
```
Unless `'0'` string is forbidden as well

Reply:
Strictly speaking, the secret must be at least 256 bits because the `sha256` function is used.
But here I only wanted to hide the IP address and username-IP. The use of a secret is to avoid the simple `sha256` hashing function and prevent recovery e.g. using a rainbow table. So all non-empty strings are fine I think, including `0`.
From the perspective of `SecurityBundle`, `0` is very unlikely because the secret of the application is injected.